src/main/msp/msp.c

```diff
@@ -2318,20 +2381,121 @@ static mspResult_e mspProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 
             if (sbufBytesRemaining(src) >= 2) {
                 vtxSettingsConfigMutable()->power = sbufReadU8(src);
+                const uint8_t newPitmode = sbufReadU8(src);
                 if (vtxType != VTXDEV_UNKNOWN) {
                     // Delegate pitmode to vtx directly
-                    const uint8_t newPitmode = sbufReadU8(src);
                     unsigned vtxCurrentStatus;
                     vtxCommonGetStatus(vtxDevice, &vtxCurrentStatus);
                     if ((bool)(vtxCurrentStatus & VTX_STATUS_PIT_MODE) != (bool)newPitmode) {
                         vtxCommonSetPitMode(vtxDevice, newPitmode);
                     }
+                }
+            }
 
-                    if (sbufBytesRemaining(src)) {
-                        vtxSettingsConfigMutable()->lowPowerDisarm = sbufReadU8(src);
-                    }
+            if (sbufBytesRemaining(src)) {
+                    vtxSettingsConfigMutable()->lowPowerDisarm = sbufReadU8(src);
+            }
+
+            // API version 1.42
+            if (sbufBytesRemaining(src) >= 9) {
+                vtxSettingsConfigMutable()->pitModeFreq = sbufReadU16(src);
+#ifdef USE_VTX_TABLE
+                const uint8_t newBandCount = sbufReadU8(src);
+                const uint8_t newChannelCount = sbufReadU8(src);
+                const uint8_t newPowerCount = sbufReadU8(src);
+
+                if ((newBandCount > VTX_TABLE_MAX_BANDS) ||
+                    (newChannelCount > VTX_TABLE_MAX_CHANNELS) ||
+                    (newPowerCount > VTX_TABLE_MAX_POWER_LEVELS)) {
+                    return MSP_RESULT_ERROR;
+                }
+                vtxTableConfigMutable()->bands = newBandCount;
+                vtxTableConfigMutable()->channels = newChannelCount;
+                vtxTableConfigMutable()->powerLevels = newPowerCount;
+
+                // We expect that the detailed band/channel and power level messages will
+                // follow to repopulate the tables, so clear them in anticipation.
+                for (int i = 0; i < VTX_TABLE_MAX_BANDS; i++) {
+                    vtxTableConfigClearBand(vtxTableConfigMutable(), i);
+                    vtxTableConfigClearChannels(vtxTableConfigMutable(), i, 0);
+                }
+                vtxTableConfigClearPowerLabels(vtxTableConfigMutable(), 0);
+                vtxTableConfigClearPowerValues(vtxTableConfigMutable(), 0);
+#else
+                sbufReadU8();
+                sbufReadU8();
+                sbufReadU8();
+#endif
+                // Added standalone values for band, channel and frequency to move
+                // away from the flawed encoded combined method originally implemented.
+                vtxSettingsConfigMutable()->band = sbufReadU8(src);
```

Comment:
Yes it would. They were added later after the initial fields so they were just tacked on. Since I'm already going to make other changes to the `MSP_SET_VTX_CONFIG` message I'll reorder. Sorry @McGiverGim.